### PR TITLE
Issue 11996: Not skip duplicate steps

### PR DESCRIPTION
**Pull request by perillai**
## Proposed changes

[//]: Fix for the issue [11996](https://github.com/webdriverio/webdriverio/issues/11996)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by christian-bromann** [comment]
@perillai thanks for the PR, seems like a unit test snapsot makes the test fail, can you verify and update the snapshot?

File: packages/wdio-spec-reporter/tests/__snapshots__/index.test.ts.snap [hunk 1]
```diff
@@ -157,9 +157,13 @@ exports[`SpecReporter > onRetry > should group retried test cases 1`] = `
 [loremipsum 50 Windows 10 #0-0] Foo test
 [loremipsum 50 Windows 10 #0-0]    green ✓ foo
 [loremipsum 50 Windows 10 #0-0]    green ✓ bar
+[loremipsum 50 Windows 10 #0-0]    red ✖ loo
 [loremipsum 50 Windows 10 #0-0]    green ✓ loo (1 retries)
```
Comment (christian-bromann) [inline packages/wdio-spec-reporter/tests/__snapshots__/index.test.ts.snap:161]: This is not desired. If a test gets retried the failing tests before should not be displayed.

**Review by christian-bromann — COMMENTED** [review]
Can we revert the formatting changes first, it makes it hard to review actual changes.

File: packages/wdio-spec-reporter/tests/__snapshots__/index.test.ts.snap (hunk 1 above)
Comment (perillai) [reply packages/wdio-spec-reporter/tests/__snapshots__/index.test.ts.snap:161]: The retries come from specFileRetries in wdio config. Though it is initialised as 0 in test:start, while test ends, it is not updated but rather updated 'retries' is passed to onRunnerEnd which is not updating the retries in hooksAndTests. I ran with old original code and still (1 retries) is not getting printed. I tried with both specFileRetries and retry under cucumberOpts.

**Pull request merged**